src/login/exchangeCodeForToken.ts

```diff
@@ -0,0 +1,56 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Environment } from "@azure/ms-rest-azure-env";
+import { Disposable, EventEmitter, Uri, UriHandler, window } from "vscode";
+import { AuthProviderBase } from "./AuthProviderBase";
+
+class UriEventHandler extends EventEmitter<Uri> implements UriHandler {
+	public handleUri(uri: Uri): void {
+		this.fire(uri);
+	}
+}
+
+const handler: UriEventHandler = new UriEventHandler();
+window.registerUriHandler(handler);
```

Comment:
I don't want us executing any code at the file-level like this. There's a risk of timing issues and any errors that occur would not be handled well (no telemetry, etc.). Instead, it should happen within the `activate` method or at least part of the `activate` call stack

Also, technically the result of `registerUriHandler` should be added to `ext.context.subscriptions` since it's a Disposable.